OloEngine/src/OloEngine/Audio/AudioSource.cpp

```
@@ -28,38 +28,198 @@ namespace OloEngine
+        if (handle == Audio::kInvalidVoiceHandle)
+        {
+            // Budget unavailable (only possible if Acquire was handed a null host, which
+            // cannot happen here) — fall back to unmanaged playback rather than silence.
+            ::ma_sound_seek_to_pcm_frame(m_Sound.get(), 0);
+            ::ma_sound_start(m_Sound.get());
+        }
+        // Otherwise Acquire has already driven OnVoiceStart if this voice won a slot;
+        // if it did not, the source is virtual and starts when one frees.
     }
 
     void AudioSource::Pause() const
```

Cppcheck (2.21.0)

[style] The function 'Pause' is never used.

(unusedFunction)